src/scripts/main.js

```diff
@@ -1,3 +1,329 @@
+
+      select.options[cityIndex].selected = true;
+      cell.innerText = '';
+      select.classList.add('cell-input');
+      cell.append(select);
+      cell.contentEditable = false;
+
+      cell.addEventListener('change', () => {
+        cell.innerText = cell.firstElementChild.value;
+
+        pushNotification(
+          10,
+          10,
+          `${tbody.headName}`,
+          `New ${tbody.headName} is added to table.\n `,
+          'success'
+        );
+      });
+    }
+
+    cell.addEventListener('blur', () => {
+      const content = cell.innerText.trim();
+
+      if (columnNum === 4 && content.slice(0, 1) !== '$') {
+        if (!isNaN(parseFloat(content))) {
+          cell.innerText = `$${(+content).toLocaleString('en')}`;
+
+          pushNotification(
+            10,
+            10,
+            `${tbody.headName}`,
+            `New ${tbody.headName} is added to table.\n `,
+            'success'
+          );
+        } else {
+          cell.innerText = defaultValue;
+
+          pushNotification(
+            10,
+            10,
+            `${tbody.headName}`,
+            `New ${tbody.headName} is not correct.\n `,
+            'error'
+          );
+        }
+      }
+
+      if (columnNum === 3) {
+        if (+content > 90
+            || +content < 18
+            || isNaN(content)) {
+          cell.innerText = defaultValue;
+
+          pushNotification(
+            10,
+            10,
+            `${tbody.headName}`,
+            `New ${tbody.headName} is not correct.\n `,
+            'error'
+          );
+        } else if (defaultValue !== content) {
+          pushNotification(
+            10,
+            10,
+            `${tbody.headName}`,
+            `New ${tbody.headName} is added to table.\n `,
+            'success'
+          );
+        }
+      }
+
+      if (columnNum === 0 || columnNum === 1) {
+        if (!isNaN(parseFloat(content))) {
+          cell.innerText = defaultValue;
+
+          pushNotification(
+            10,
+            10,
+            `${tbody.headName}`,
+            `New ${tbody.headName} string is not correct.\n `,
+            'error'
+          );
+        } else if (defaultValue !== content && content) {
+          pushNotification(
+            10,
+            10,
+            `${tbody.headName}`,
+            `New ${tbody.headName} is added to table.\n `,
+            'success'
+          );
+        }
+      }
+
+      if (!content) {
+        cell.innerText = defaultValue;
+      }
+    });
+
+    cell.addEventListener('keydown', eventCell => {
+      if (eventCell.key === 'Enter') {
+        cell.blur();
+      };
+    });
+  }
+});
+
+function createInput(labelText, inputName, data, inputtype = 'text') {
+  const label = document.createElement('LABEL');
+
+  label.innerHTML = `${labelText}`
+                    + `<input name =${inputName} `
+                    + `type=${inputtype} `
+                    + `data-qa=${data} required></input>`;
```

Comment:
interpolation allow spaces
```suggestion
  label.innerHTML = `${labelText}
                                    <input 
                                      name =${inputName}
                                      type=${inputtype}
                                      data-qa=${data} 
                                      required
                                    >
                                    </input>
                                  `;
```